monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -303,22 +306,62 @@ class BundleGen(AlgoGen):
         python -m monai.apps.auto3dseg BundleGen generate --data_stats_filename="../algorithms/data_stats.yaml"
     """
 
-    def __init__(self, algo_path: str = ".", algos=None, data_stats_filename=None, data_src_cfg_name=None):
-        self.algos: Any = []
-
-        if algos is None or isinstance(algos, str):
-            # trigger the download process
-            zip_download_dir = TemporaryDirectory()
-            algo_compressed_file = os.path.join(zip_download_dir.name, "algo_templates.tar.gz")
-            download_and_extract(default_algo_zip if algos is None else algos, algo_compressed_file, algo_path)
-            zip_download_dir.cleanup()
-            sys.path.insert(0, os.path.join(algo_path, "algorithm_templates"))
-            algos = deepcopy(default_algos)
-            for name in algos:
-                algos[name]["template_path"] = os.path.join(
-                    algo_path, "algorithm_templates", algos[name]["template_path"]
-                )
+    def __init__(
+        self,
+        algo_path: str = ".",
+        algos=None,
+        templates_path_or_url=None,
+        data_stats_filename=None,
+        data_src_cfg_name=None,
+    ):
+
+        if algos is None or isinstance(algos, (list, tuple, str)):
```

Comment:
Since you are using `templates_path_or_url` for the url, user do not need to put the url in the `algos` anymore. I would suggest to remove ` str` here in instance check.

Reply:
the string here is for algo name (not url) .  so a user can specify `algos=['segresnet', 'dints'] or algos='segresnet'`